core/src/banking_stage.rs

```diff
@@ -300,7 +328,22 @@ impl BankingStage {
                     Ok(())
                 }
             }
-            _ => Ok(()),
+            _ => {
+                // When holding packets, do some deserialization in the downtime
+                // to get ready for leader slot.
+                let mut times = 0;
+                for (packets, transactions, packet_indexes) in buffered_packets.iter_mut() {
+                    if transactions.is_none() {
+                        *transactions =
+                            Some(Self::transactions_from_packets(&packets, &packet_indexes));
+                        times += 1;
+                    }
+                    if times > 2 {
```

Comment:
Is there any specific reason for this number? Can we also make it a constant?

Reply:
Worked well in terms of not creating too much latency for checking if the bank is available and produced enough deserialized packets for the start of the slot.